Description: work around clang hanging forever when trying to optimize the sqlite3 build
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1878292

--- a/third_party/sqlite3/src/moz.build
+++ b/third_party/sqlite3/src/moz.build
@@ -98,3 +98,8 @@ if CONFIG['CC_TYPE'] in ('clang', 'gcc')
         '-Wno-sign-compare',
         '-Wno-type-limits',
     ]
+
+# Work around clang hanging forever when trying to optimize the sqlite3 build
+# with the default "-O2" parameter. See https://launchpad.net/bugs/1878292.
+if CONFIG['CC_TYPE'] == 'clang' and CONFIG['CPU_ARCH'] == 's390x':
+    CFLAGS += ['-O0']
